modules/dnn/test/test_model.cpp

```diff
@@ -0,0 +1,163 @@
+
+    Size size{227, 227};
+    float norm = 1e-4;
+
+    testClassifyModel(weights_file, config_file, img_path, ref, norm, size);
+}
+
+
+TEST_P(Test_Model, DetectRegion)
+{
+    std::vector<int> refClassIds = {6, 1, 11};
+    std::vector<float> refConfidences = {0.750469f, 0.780879f, 0.901615f};
+    std::vector<Rect2d> refBoxes = {Rect2d(240.032, 52.994, 135.408, 72.134),
+                                    Rect2d(112.320, 109.865, 192.067, 200.220),
+                                    Rect2d(57.658, 140.608, 117.603, 249.683)};
+
+    std::string img_path = _tf("dog416.png");
+    std::string weights_file = _tf("yolo-voc.weights");
+    std::string config_file = _tf("yolo-voc.cfg");
+
+    float scale = 1.0 / 255.0;
+    Size size{416, 416};
+    bool swapRB = true;
+
+    double confThreshold = 0.24;
+    double scoreDiff = (target == DNN_TARGET_OPENCL_FP16 || target == DNN_TARGET_MYRIAD) ? 1e-2 : 8e-5;
+    double iouDiff = 0.014;
+    double nmsThreshold = (target == DNN_TARGET_MYRIAD) ? 0.397 : 0.4;
+
+    testDetectModel(weights_file, config_file, img_path, refClassIds, refConfidences,
+                    refBoxes, scoreDiff, iouDiff, confThreshold, nmsThreshold, size,
+                    Scalar(), scale, swapRB);
+}
+
+TEST_P(Test_Model, DetectionOutput)
+{
+    std::vector<int> refClassIds = {7, 12};
+    std::vector<float> refConfidences = {0.991359f, 0.94786f};
+    std::vector<Rect2d> refBoxes = {Rect2d(491.822, 81.1668, 211.751, 98.0672),
+                                    Rect2d(132.093, 223.903, 206.984, 343.257)};
+
+    std::string img_path = _tf("dog416.png");
+    std::string weights_file = _tf("resnet50_rfcn_final.caffemodel");
+    std::string config_file = _tf("rfcn_pascal_voc_resnet50.prototxt");
+
+    Scalar mean = Scalar(102.9801, 115.9465, 122.7717);
+    Size size{800, 600};
+
+    double scoreDiff = (backend == DNN_BACKEND_OPENCV && target == DNN_TARGET_OPENCL_FP16) ?
+                        4e-3 : default_l1;
+    double iouDiff = 0.016;
+    float confThreshold = 0.8;
+    float nmsThreshold = 0;
+
+    testDetectModel(weights_file, config_file, img_path, refClassIds, refConfidences, refBoxes,
+                    scoreDiff, iouDiff, confThreshold, nmsThreshold, size, mean);
+}
+
+
+TEST_P(Test_Model, DetectionMobilenetSSD)
+{
+    std::vector<int> refClassIds = {7, 15};
+    std::vector<float> refConfidences = {0.99983513f, 0.87925464f};
+    std::vector<Rect2d> refBoxes = {Rect2d(329, 239, 85, 102),
+                                    Rect2d(101, 189, 34, 138)};
+
+    std::string img_path = _tf("street.png");
+    std::string weights_file = _tf("MobileNetSSD_deploy.caffemodel");
+    std::string config_file = _tf("MobileNetSSD_deploy.prototxt");
+
+    Scalar mean = Scalar(127.5, 127.5, 127.5);
+    float scale = 1.0f / 127.5;
+    Size size{300, 300};
+
+    double scoreDiff = (target == DNN_TARGET_OPENCL_FP16 || target == DNN_TARGET_MYRIAD) ? 1.5e-2 : 1e-5;
+    double iouDiff = 0.0098;
```

Comment:
Please check if we can use different IoU thresholds for FP32 and FP16 targets. In the origin test from https://github.com/opencv/opencv/blob/master/modules/dnn/test/test_caffe_importer.cpp there were:
```cpp
    const float scores_diff = (targetId == DNN_TARGET_OPENCL_FP16 || targetId == DNN_TARGET_MYRIAD) ? 1.5e-2 : 1e-5;
    const float boxes_iou_diff = (targetId == DNN_TARGET_OPENCL_FP16 || targetId == DNN_TARGET_MYRIAD) ? 6.3e-2 : 1e-4;
```